### ISSUE TITLE
Type of statistics.{min, max} is expected to be Uint8Array, but is string

### ISSUE THREAD

**Issue by severo** [hyparam/hyparquet#8]
The type of `Statistics.max` should be `Uint8Array`:

https://github.com/hyparam/hyparquet/blob/938076b3bc37cc27018d27618a0d392a2768375d/src/types.d.ts#L179

But the value is a string:

<img width="630" alt="Capture d’écran 2024-05-02 à 17 59 50" src="https://github.com/hyparam/hyparquet/assets/1676121/205240d1-b3ae-44dd-a94c-98e6b1f888bd">

See https://observablehq.com/@huggingface/hyparquet-uint8array-value for a reproduction.

**Comment by severo** [comment]
Additional question: why not return a number?

**Comment by platypii** [comment]
@severo you're right, that type definition should be a `string`, nice catch! Pushing a fix now.

### Why its a string:

Metadata parsing follows the [parquet thrift schema](https://github.com/apache/parquet-format/blob/master/src/main/thrift/parquet.thrift#L258-L259) which defines the min/max as a `binary` type which is a [bytearray/string type in thrift](https://thrift.apache.org/docs/types#special-types).

I wanted the metadata to be JSON serializable as much as possible, so I encode the byte array as a utf8 string.

In the case of a numeric column, this results in ugly strings as shown above. But parquet columns can also contain string data as the min/max. And so string seemed like the best type to use to encode general binary data.

**Comment by severo** [comment]
OK, thanks for fixing the type in https://github.com/hyparam/hyparquet/commit/f411b83f5e3c22d1820d1f073c8555ea02f38201. But the underlying question is: how should I consider the value when the column contains numbers? How can I display it as a string?

I can get the Uint8Array back with:

```javascript
const maxValue = metadata.row_groups[0].columns[2].meta_data.statistics.max;
console.log(maxValue)
/// ^ "�'\u0000\u0000\u0000\u0000\u0000\u0000"
const encoder = new TextEncoder()
console.log(encoder.encode())
/// ^ Uint8Array(10) [ 239, 191, 189, 39, 0, 0, 0, 0, 0, 0 ]
```

But I have no idea how to convert this back to a number. From the column metadata, the type is `row_groups.0.columns.2.meta_data.type: "INT64"`. I imagine that each type needs its own conversion function to JavaScript types, no?

**Comment by platypii** [comment]
I think you need to call `.getBigInt64(offset, true)` with `littleEndian = true`.

I'm looking into adding a conversion step to the metadata parsing, to convert Statistics from string to the appropriate js type. Will be helpful for making filtered queries. Will keep you posted.

**Comment by platypii** [comment]
Just published v0.9.0 which includes:
 - parsing of min/max as native javascript types
 - additional metadata that wasn't being parsed before
 - update types to accurately reflect bigint vs number

I changed thrift.js to return a `Uint8Array` and then in metadata.js handle conversion to `bigint | boolean | number | string` as appropriate for the column type.

Good catch! Thanks for reporting

**Comment by severo** [comment]
Awesome, it works perfectly!
